### PR TITLE
OcBootManagementLib: Revert changes to fat slice secure boot logic from 125d8e8df36bd7ca12085d9670dca9060cf49c76

### PR DESCRIPTION
**Pull request by mikebeaton**
Note: EFI_SUCCESS does not always mean a signature was found and verified (which cannot happen for a fat slice), but simply that no further verification of SB status is required and OC image loader should be used.

Resolves: https://github.com/acidanthera/bugtracker/issues/2414

### CODE REVIEW

File: Library/OcBootManagementLib/ImageLoader.c [hunk 1]
```diff
@@ -969,7 +969,7 @@ InternalEfiLoadImage (
   // Load the image ourselves in secure boot mode.
   //
   if (SecureBootStatus == EFI_SUCCESS) {
-    if ((SourceBuffer != NULL) && (OcImageLoaderCaps == NULL)) {
+    if (SourceBuffer != NULL) {
```
Comment (vit9696) [inline Library/OcBootManagementLib/ImageLoader.c:972]: Do you remember what was the purpose for `OcImageLoaderCaps == NULL` check?

File: Library/OcBootManagementLib/ImageLoader.c [hunk 2]
```diff
@@ -980,10 +980,9 @@ InternalEfiLoadImage (
                  );
     } else {
       //
-      // We verified the image, but contained garbage, or we are trying to secure boot a Fat slice.
-      // This should not happen.
+      // We verified the image, but contained garbage.
+      // This should not happen, just abort.
       //
-      ASSERT (FALSE);
```
Comment (vit9696) [inline Library/OcBootManagementLib/ImageLoader.c:986]: Why did you remove the `ASSERT`? Is this code reachable from now on?

File: Library/OcBootManagementLib/ImageLoader.c (hunk 1 above)
Comment (mikebeaton) [reply Library/OcBootManagementLib/ImageLoader.c:972]: Yes. Go to the commit I'm (proposing) reverting a small part of: https://github.com/acidanthera/OpenCorePkg/commit/125d8e8df36bd7ca12085d9670dca9060cf49c76

And forward search in the page for 'fat slice'. This leads to changes I made (see lines 944 and 958 of the revised ImageLoader.c file in the result of that diff). I am reverting to your original code.

I believe I added the fat slice check because it seemed to me (wrongly, it now turns out) that there was no way that a fat slice could sanely have success in secure boot status (since they date from before signing).

File: Library/OcBootManagementLib/ImageLoader.c (hunk 2 above)
Comment (mikebeaton) [reply Library/OcBootManagementLib/ImageLoader.c:986]: I am removing the assert because I added it - I don't think would do any harm to keep it, however. I can say with fair certainty that I added it so that if my added check ([see above](https://github.com/acidanthera/OpenCorePkg/pull/548#discussion_r1668144594)) was wrong, it would get flagged up sooner rather than later and be easy to identify...

File: Library/OcBootManagementLib/ImageLoader.c (hunk 1 above)
Comment (vit9696) [reply Library/OcBootManagementLib/ImageLoader.c:972]: Got it, ok.

File: Library/OcBootManagementLib/ImageLoader.c (hunk 2 above)
Comment (vit9696) [reply Library/OcBootManagementLib/ImageLoader.c:986]: Just perhaps if it is an invariant, we should replace `Status` assignment with a `CpuDeadLoop()` call and keep the assert. If it is not, then your change makes sense.

Comment (mikebeaton) [reply Library/OcBootManagementLib/ImageLoader.c:986]: I don't think it is an invariant - we could get get to here without really verifying the file (when it's a dmg, as we are seeing in this issue), but then the file could generate an error when extracting a fat slice.

Comment (vit9696) [reply Library/OcBootManagementLib/ImageLoader.c:986]: Ok

File: Library/OcBootManagementLib/ImageLoader.c [hunk 3]
```diff
@@ -980,10 +980,9 @@ InternalEfiLoadImage (
                  );
     } else {
       //
-      // We verified the image, but contained garbage, or we are trying to secure boot a Fat slice.
-      // This should not happen.
+      // We verified the image, but contained garbage.
+      // This should not happen, just abort.
```
Comment (vit9696) [inline Library/OcBootManagementLib/ImageLoader.c:984]: Reword this comment that this can happen when a trusted binary failed to load for some reason.

Comment (mikebeaton) [reply Library/OcBootManagementLib/ImageLoader.c:984]: I agree, was just thinking up some wording.

File: Library/OcBootManagementLib/ImageLoader.c [hunk 4]
```diff
@@ -980,10 +980,16 @@ InternalEfiLoadImage (
...
-      // This should not happen.
+      // SecureBootStatus can be EFI_SUCCESS without a signature being
+      // verified, if SecureBootModel is Disabled and the boot file is from
+      // a DMG. This effectively just means: no more SB checks required, use
+      // our loader.
+      // SourceBuffer will be NULL if filtering for supported architecture
+      // failed.
+      // All this can't happen in a secure boot chain, and should not happen
+      // (with valid boot files) in a non-secure boot chain. But if it does,
+      // just abort.
```
Comment (mikebeaton) [inline Library/OcBootManagementLib/ImageLoader.c:991]: Is this about right?

Comment (vit9696) [reply Library/OcBootManagementLib/ImageLoader.c:991]: Yes
